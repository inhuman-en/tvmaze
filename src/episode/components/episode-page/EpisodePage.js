import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import './EpisodePage.scss';

import { ErrorMessage, Spinner } from 'shared';

import { loadEpisodeDetails } from '../../store';
import { useEpisode } from '../../hooks';

import EpisodeDetails from '../episode-details/EpisodeDetails';

const EpisodePage = () => {
  const { id } = useParams();
  const { data, error } = useEpisode();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadEpisodeDetails(id));
  }, [id, dispatch]);

  if (error) {
    return <ErrorMessage text={error} />;
  }

  if (!data) {
    return <Spinner />;
  }

  return <EpisodeDetails data={data} />;
};

export default EpisodePage;
