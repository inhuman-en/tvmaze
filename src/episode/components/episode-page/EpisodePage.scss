.episode-page {
  &__main {
    margin-top: 30px;
  }
}