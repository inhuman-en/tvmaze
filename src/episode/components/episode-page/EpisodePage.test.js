import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EpisodePage from './EpisodePage';

configure({ adapter: new Adapter() });

jest.mock('../../hooks');

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: 2 }),
}));
jest.mock('react-redux');

describe('EpisodePage component', () => {
  it('should render', () => {
    const result = toJson(shallow(<EpisodePage />));
    expect(result).toMatchSnapshot();
  });
});
