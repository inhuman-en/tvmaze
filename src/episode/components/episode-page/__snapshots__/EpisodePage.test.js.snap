// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage component should render 1`] = `
<div
  className="episode-page__main"
>
  <EpisodeDetails
    data={
      Object {
        "image": Object {
          "medium": "https://example.com/medium",
          "original": "https://example.com/original",
        },
        "name": "episode name",
        "summary": "episode summary",
      }
    }
  />
</div>
`;
